webapp/src/components/cardDetail/cardDetailContents.test.tsx

```diff
@@ -0,0 +1,258 @@
+    range.getBoundingClientRect = jest.fn()
+
+    range.getClientRects = () => {
+        return {
+            item: () => null,
+            length: 0,
+            [Symbol.iterator]: jest.fn(),
+        }
+    }
+    return range
+}
+window.focus = jest.fn()
+
+describe('components/cardDetail/cardDetailContents', () => {
+    const board = TestBlockFactory.createBoard()
+    board.fields.cardProperties = [
+        {
+            id: 'property_id_1',
+            name: 'Owner',
+            type: 'select',
+            options: [
+                {
+                    color: 'propColorDefault',
+                    id: 'property_value_id_1',
+                    value: 'Jean-Luc Picard',
+                },
+                {
+                    color: 'propColorDefault',
+                    id: 'property_value_id_2',
+                    value: 'William Riker',
+                },
+                {
+                    color: 'propColorDefault',
+                    id: 'property_value_id_3',
+                    value: 'Deanna Troi',
+                },
+            ],
+        },
+    ]
+
+    const card = TestBlockFactory.createCard(board)
+
+    test('should match snapshot', async () => {
+        const component = wrapProviders((
+            <CardDetailContents
+                id='test-id'
+                card={card}
+                contents={[]}
+                readonly={false}
+            />
+        ))
+
+        let container: Element | undefined
+        await act(async () => {
+            const result = render(component)
+            container = result.container
+        })
+        expect(container).toMatchSnapshot()
+    })
+
+    test('should match snapshot with contents array', async () => {
+        const contents = [TestBlockFactory.createDivider(card)]
+        const component = wrapProviders((
+            <CardDetailContents
+                id='test-id'
+                card={card}
+                contents={contents}
+                readonly={false}
+            />
+        ))
+
+        let container: Element | undefined
+        await act(async () => {
+            const result = render(component)
+            container = result.container
+        })
+        expect(container).toMatchSnapshot()
+    })
+
+    test('should match snapshot after onBlur triggers', async () => {
+        const component = wrapProviders((
+            <CardDetailContents
+                id='test-id'
+                card={card}
+                contents={[]}
+                readonly={false}
+            />
+        ))
+
+        let container: Element | undefined
+        await act(async () => {
+            const result = render(component)
+            container = result.container
+        })
+        const markdownEditorField = container!.querySelectorAll('.octo-editor-preview.octo-placeholder')
+        expect(markdownEditorField).toBeDefined()
+        fireEvent.click(markdownEditorField[markdownEditorField.length - 1]!)
```

Comment:
Why the last element of `markdownEditorField` is used? Should we expect to have only one element in this NodeList?

Reply:
I did this on my own through web browser and tested what elements are present. There are several that come up with that class so and the last one happens to be the content space I was looking for

Reply:
This is really strange. Here is how it looks in the debugger:
![image](https://user-images.githubusercontent.com/5587620/128397667-df263d6f-c3ca-4a88-a21e-5dfcd2e910fc.png)
Also in the snapshot there is only one element matching the query.

Reply:
Oh cool! I didn't notice that in the test. Not a big deal but I switch from querySelectorAll to just querySelector. Thanks